### PR TITLE
feat: update sync logic to fire on node selection

### PR DESCRIPTION
**Pull request by bheston**
# Description

Updated sync logic to fire on node selection. This will update a component when it's added or pasted into a new container.

## Motivation & context

Currently you could have a themed design, but when adding a new theme-aware component it would not update to the theme settings. With this change it will update immediately when being placed into a themed container.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
Going to pass on reviewing this since I'm not particularly familiar with this area of the codebase. What's the chance I could get a rundown on the Figma plugin sometime next week to bring me up to speed?

**Comment by bheston** [comment]
@EisenbergEffect no worries. I could cover the "what" of the Figma plugin, but @nicholasrice or @Jeremy-Knudsen know the code better.

File: packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts [hunk 1]
```diff
@@ -1,8 +1,34 @@
...
+import { PluginNodeData } from "../core/node";
 import { PluginUIProps } from "../core/ui";
-import { FigmaPluginNode } from "./node";
+import { canHaveChildren, FigmaPluginNode, isInstanceNode } from "./node";
 
 export class FigmaController extends Controller {
+    private syncInstanceWithMaster(target: InstanceNode): void {
+        const source = target.masterComponent;
+
+        function sync(_source: BaseNode, _target: BaseNode): void {
```
Comment (nicholasrice) [inline packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts:10]: You probably don't need a `sync` function - this work can just go in the body of `syncInstanceWithMaster`

File: packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts [hunk 2]
```diff
@@ -11,6 +37,15 @@ export class FigmaController extends Controller {
         }
     }
 
+    public sync(nodeIds: string[]): void {
```
Comment (nicholasrice) [inline packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts:40]: Can you add comments describing what "sync" is intended to do?

**Comment by EisenbergEffect** [comment]
This PR is getting a little stale. @bheston Can you reach out to a few targeted individuals who you think can help with the review?

File: packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts (hunk 2 above)
Comment (bheston) [reply packages/tooling/fast-figma-plugin-msft/src/figma/controller.ts:40]: Added at the generic controller level.

**Pull request merged**